core-blocks/video/edit.js

```diff
@@ -35,6 +37,23 @@ class VideoEdit extends Component {
 		this.onSelectURL = this.onSelectURL.bind( this );
 	}
 
+	componentDidMount() {
+		const { attributes, setAttributes } = this.props;
+		const { id, src = '' } = attributes;
+		if ( ! id && src.indexOf( 'blob:' ) === 0 ) {
+			const file = getBlobByURL( src );
+			if ( file ) {
+				editorMediaUpload( {
+					filesList: [ file ],
+					onFileChange: ( [ { url } ] ) => {
+						setAttributes( { src: url } );
+					},
+					allowedType: 'video',
```

Comment:
We'll need to handle any errors that happen during the upload here using `onError`. See https://github.com/WordPress/gutenberg/pull/8066 for an example that fixes the File block.